### ISSUE TITLE
Pyright flags an error when passing the result of `dict.fromkeys` to `dict.update` when the dict being updated has a Union key type

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2537]
When passing the result of `dict.fromkeys` to `dict.update` when the key type of the dict is a union, pyright flags an error while mypy doesn't. I believe this is because mypy infers the type of the returned dict based on the type of `options` so they match.

**Screenshots or Code**
```python
from typing import Union

options: dict[Union[int, str], int] = {}
channel_types: dict[str, int] = {}

reveal_type(channel_types)
reveal_type(dict.fromkeys(channel_types.keys(), 1))

options.update(channel_types) # error on both mypy and pyright
options.update(dict.fromkeys(channel_types.keys(), 1)) # error on pyright only
```
Pyright output:
```python
info: Type of "channel_types" is "dict[str, int]"
info: Type of "dict.fromkeys(channel_types.keys(), 1)" is "dict[str, int]"
error: Argument of type "dict[str, int]" cannot be assigned to parameter "__m" of type "Iterable[tuple[int | str, int]]" in function "update"
    TypeVar "_T_co@Iterable" is covariant
      "str" is incompatible with "tuple[int | str, int]" (reportGeneralTypeIssues)
error: Argument of type "dict[str, int]" cannot be assigned to parameter "__m" of type "Iterable[tuple[int | str, int]]" in function "update"
    TypeVar "_T_co@Iterable" is covariant
      "str" is incompatible with "tuple[int | str, int]"
```
Mypy output:
```python
note: Revealed type is "builtins.dict[builtins.str, builtins.int]"
note: Revealed type is "builtins.dict[builtins.str*, builtins.int*]"
error: Argument 1 to "update" of "dict" has incompatible type "Dict[str, int]"; expected "Mapping[Union[int, str], int]"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (pyright 1.1.184)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.185, which I just published. It will also be included in the next release of pylance.
